diff --git a/node_modules/@puppeteer/browsers/lib/esm/install.js b/node_modules/@puppeteer/browsers/lib/esm/install.js
index cf6db62..b2df977 100644
--- a/node_modules/@puppeteer/browsers/lib/esm/install.js
+++ b/node_modules/@puppeteer/browsers/lib/esm/install.js
@@ -128,7 +128,7 @@ async function installUrl(url, options) {
     if (downloadProgressCallback === 'default') {
         downloadProgressCallback = await makeProgressCallback(options.browser, options.buildIdAlias ?? options.buildId);
     }
-    const fileName = url.toString().split('/').pop();
+    const fileName = decodeURIComponent(url.toString().split('/').pop());
     assert(fileName, `A malformed download URL was found: ${url}.`);
     const cache = new Cache(options.cacheDir);
     const browserRoot = cache.browserRoot(options.browser);
